src/components/TxData.js

```diff
@@ -61,6 +63,14 @@ class TxData extends React.Component {
 
   componentDidUpdate = (prevProps) => {
     if (prevProps.transaction !== this.props.transaction) {
+      // When we are changing the transaction but keeping the same component
```

Comment:
I agree we must clean the state. But I feel that we should rever reuse a component. We should always create a new one (and it must be done by whoever is using it). A simple way to do it is to set a `<TxData ... key={txId} />` because it will prevent react from matching the same component with another transaction.

Reply:
I agree with you! Just updated the code

Reply:
Great!